projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx

```diff
@@ -430,11 +437,22 @@ export default function PosterPanel( {
 
 	const onPreviewOnHoverChange = useCallback(
 		( shouldPreviewOnHover: boolean ) => {
+			let newPosterData: PosterDataProps = {
+				...attributes.posterData,
+				previewOnHover: shouldPreviewOnHover,
+			};
+
+			// Add default values for the preview options on activation
+			if ( shouldPreviewOnHover ) {
+				newPosterData = {
+					previewAtTime,
+					previewLoopDuration,
+					...newPosterData,
+				};
+			}
```

Comment:
I'm still trying to figure out this part. The fundamental issue is the block doesn't provide default values. We can add them to the block definition in the blocks.json file. Something like...

```json
		"posterData": {
			"type": "object",
			"default": {
				"previewOnHover": false,
				"previewAtTime": 0,
				"previewLoopDuration": 5000
			}
		},
```

Reply:
> block doesn't provide default values

It does:
https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx#L405-L407

Reply:
[These default values](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/poster-panel/index.tsx#L405-L407) are handled at `< PosterPanel />` component level, but the `posterData` block attribute doesn't have values when it's just created.
So, out of the PosterPanel component context, `posterData` will be undefined.

Reply:
> but the posterData block attribute doesn't have values when it's just created.

When `previewOnHover` is enabled, the `posterData` object is set to `{ previewOnHover: true }`.
What this change does is add the default values when enabling it if those are not already defined (they are added before the `...newPosterData`). It will not be undefined.